packages/formation-react/src/components/MediaLink/MediaLink.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { createIcon } from '../../helpers/link-utils';
+
+// List of known media sites
+export const MEDIA_SITES = {
+  'youtube.com': {
+    name: 'YouTube',
+    icon: 'fab fa-youtube',
+    type: 'video',
+  },
+  'instagram.com': {
+    name: 'Instagram',
+    icon: 'fab fa-instagram',
+    type: 'images',
+  },
+};
+
+/**
+ * Determine if the url contains the url of a known video hosting site
+ * @param {string} url - url to video hosting site
+ * @return {boolean} - boolean value indicating that the site is a known host
+ */
+export const findMediaSite = url =>
+  Object.keys(MEDIA_SITES).find(site =>
+    (url || '').toLowerCase().includes(site),
+  );
+
+/**
+ * Object passed to the media link contents render function
+ * @typedef MediaLink~linkProps
+ * @type {Object.<string, *>}
+ * @property {string} href - link href
+ * @property {string} target - set to '_blank' for external links
+ * @property {string} rel - set to 'noopener noreferrer' for external links
+
+ * @property {string} type - file MIME-type
+ * @property {...*} var_args - Include any extra parameters you want to add to the
+ *  media link, e.g. `className`, `onClick`, etc.
+ */
+/**
+ * Render media link content
+ * @function content
+ * @param {ReactElement} renderedSiteName - rendered site name, e.g.
+ *   `<dfn>on YouTube</dfn>`
+ * @param {MediaLink~props} props - all props passed into the component
+ * @param {MediaLink~linkProps} linkProps - read-only link properties
+ * @return {ReactElement} text with renderedSiteName; or customized as desired
+ */
+/**
+ * MediaLink properties
+ * @typedef MediaLink~props
+ * @type {object.<string, *>}
+ * @param {string} href (required) - url pointing to the video hosting site
+ * @param {string} title (required) - title of the file or video
+ * @param {function} content - Function that allows custom rendering of the link
+ *  content; This function returns JSX
+ * @param {string|ReactElement} icon - Fontawesome icon group + name, e.g.
+ *  "fab fa-youtube", or the icon JSX
+ * @param {boolean} external - Set to true if the link is to an external site;
+ *  setting this adds a blank target (opens a new tab) & no-referrer and opener
+ *  rel attribute
+ * @param {...*} var_args - Include any extra parameters you want to add to the
+ *  video link, e.g. `className`, etc.
+ */
+/**
+ * MediaLink component
+ * @function MediaLink
+ * @param {MediaLink~props}
+ */
+const MediaLink = props => {
+  if (!props.href) {
+    throw new Error('Media links require an href property');
+  }
+
+  const { external = true, title, content, icon = '', ...linkProps } = props;
```

Comment:
Can we determine if `external` is true automatically by comparing the `hostname` of the `href` to `window.location.hostname`? It would be easy if IE supported the `URL()` constructor. :unamused: There might be a workaround there, though; I haven't looked.

Reply:
:thinking: There might actually be a polyfill for the URL constructor. Not sure that's something we've got in our (outdated) `@babel/polyfill` import.